### PR TITLE
Fix sum batching rule, add simple clone batching rule

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47227 Batched gradient support for view+inplace operations
* #47365 Add batching rule for torch.clone(tensor, torch.contiguous_format)
* #47364 as_strided batching rule
* #47226 Batching rule for Tensor.new_empty_strided
* #47225 Implement Tensor.new_empty_strided(sizes, strides, *, dtype, device, requires_grad)
* #47223 Batched grad for advanced indexing (index)
* **#47189 Fix sum batching rule, add simple clone batching rule**
* #47188 Batching rules for complex view functions

PyTorch has a special case where sum(scalar_tensor, dim=0) does not fail
and instead returns a new copy of the original scalar_tensor. If we
end up vmapping over per-example scalar tensors, e.g.,
```
>>> x = torch.randn(B0)  # the per-examples are all scalars
>>> vmap(partial(torch.sum, dim=0), x)
```
then we should replicate the behavior of sum(scalar_tensor, dim=0) by
returning a clone of the input tensor.

This PR also adds a batching rule for clone(Tensor, MemoryFormat). The
batching rule:
- unwraps the BatchedTensor, calls clone(), and rewraps the
BatchedTensor if MemoryFormat is torch.preserve_format (which is the
default).
- errors out with an NYI for all other memory formats, including
torch.contiguous_format. There are some weird semantics for memory
layouts with vmap that I need to go and figure out. Those are noted in
the comments for `clone_batching_rule`

Test Plan:
- new tests

Differential Revision: [D24741689](https://our.internmc.facebook.com/intern/diff/D24741689)

### CODE REVIEW

File: aten/src/ATen/BatchingRegistrations.cpp [hunk 1]
```diff
@@ -648,6 +656,10 @@ TORCH_LIBRARY_IMPL(aten, Batched, m) {
   TO_BATCHING_RULE("to.other", const Tensor&, bool, bool, optional<MemoryFormat>)
   m.impl("to.dtype_layout", to_dtype_layout_batching_rule);
 #undef TO_BATCHING_RULE
+#define UNARY_POINTWISE_VA(op, ...) m.impl(#op, \
+    unary_pointwise_batching_rule<Tensor (*)(const Tensor&, __VA_ARGS__), at::op, __VA_ARGS__>);
+  UNARY_POINTWISE_VA(clone, optional<MemoryFormat>);
```
Comment (albanD) [inline aten/src/ATen/BatchingRegistrations.cpp:661]: This adds support for the clone function as well?
If so should it be tested/mentioned in the commit msg?

Comment (zou3519) [reply aten/src/ATen/BatchingRegistrations.cpp:661]: That was an accidental commit, let me remove that

Comment (zou3519) [reply aten/src/ATen/BatchingRegistrations.cpp:661]: EDIT: Yes, I should be testing clone here.

**Review by albanD — APPROVED** [review]
lgtm
